ui/apps/ui/src/app/components/sort-by-functionality/sort-by-functionality.component.ts

```diff
@@ -0,0 +1,146 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { Router } from '@angular/router';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { SEARCH_PAGE_PATH } from '@collections/services/custom-route.type';
+import { CustomRoute } from '@collections/services/custom-route.service';
+import {
+  ICollectionNavConfig,
+  ISortOption,
+} from '@collections/repositories/types';
+import { NavConfigsRepository } from '@collections/repositories/nav-configs.repository';
+import { sanitizeQuery } from '@components/search-input/query.sanitizer';
+import { filter } from 'rxjs';
+import { DefaultSortValueEnum } from '@components/sort-by-functionality/default-sort-value.enum';
+import { SortOptionValuesEnum } from '@components/sort-by-functionality/sort-option-values.enum';
+
+@UntilDestroy()
+@Component({
+  selector: 'ess-sort-by-functionality',
+  template: `<div class="col-sm-4 sort_container">
+    <label for="sorts">Sort By</label>
+    <select
+      [(ngModel)]="selectedSortOption"
+      #sorts
+      id="sorts"
+      label="Sort by"
+      class="form-select"
+      (change)="this.onSelected(sorts.value)"
+    >
+      <option *ngFor="let sortBy of sortOptions" value="{{ sortBy.value }}">
+        {{ sortBy.description }}
+      </option>
+    </select>
+  </div>`,
+  styles: [
+    `
+      :host {
+        display: contents;
+      }
+
+      label {
+        width: 100px;
+        font-family: 'Inter';
+        font-style: normal;
+        font-weight: 400;
+        font-size: 14px;
+        line-height: 20px;
+        color: #4f4f4f;
+      }
+
+      select {
+        font-family: 'Inter';
+        font-style: normal;
+        font-weight: 500;
+        font-size: 14px;
+        line-height: 24px;
+        color: #101828;
+      }
+
+      select:focus {
+        -webkit-box-shadow: none;
+        box-shadow: none;
+        border: 1px solid #ced4da;
+      }
+
+      .sort_container {
+        width: 18rem;
+        display: flex;
+        align-items: center;
+      }
+    `,
+  ],
+})
+export class SortByFunctionalityComponent implements OnInit {
+  public selectedSortOption: string = DefaultSortValueEnum.DEFAULT;
+
+  public readonly sortOptions: Array<ISortOption> = [
```

Comment:
Instead of having this array in the component I would have sort option interface which would couple caption and value. Here you have separate enums `SortOptionValuesEnum` and `DefaultSortValueEnum`. I think that having those values specified in the template itself (`<option></...`) would be simpler. This is not something that is fetched from server or calculated. So having it in template would be simpler.